net/conserver/Makefile

```
@@ -8,15 +8,15 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=conserver
-PKG_VERSION:=8.2.2
-PKG_RELEASE:=2
+PKG_VERSION:=8.2.4
+PKG_RELEASE:=1
 PKG_MAINTAINER:=Bjørn Mork <bjorn@mork.no>
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/conserver/conserver.git
 PKG_SOURCE_VERSION:=v$(PKG_VERSION)
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_MIRROR_HASH:=27d92e6c04e97cd0884774eace0b44f30087695927bcce8addc11dba9c090d7c
+PKG_MIRROR_HASH:=b95658a2c8470da32759b65ed60e9d5747da3f84ca25f3be39f8aa0c53f4cc1d
```

```suggestion
PKG_MIRROR_HASH:=8f02afb933d118fc4ccb73cda63213b76792dad4c080c7a64967a33371f8aa19
```
buildpr is failing because of this hash, and I'm also getting the same here locally.

---

Changed.  But I really need a "Figure out new PKG_MIRROR_HASH for dummies" book...

What I currently do is obviously failing for some reason I don't understand:  I simply build the new package and then run sha256sum on the generated archive in dl/:

```
bjorn@canardo:/usr/local/src/lede$ sha256sum dl/conserver-8.2.4.tar.gz 
b95658a2c8470da32759b65ed60e9d5747da3f84ca25f3be39f8aa0c53f4cc1d  dl/conserver-8.2.4.tar.gz
```

What's the proper way to calculate the hash?

---

This is correct. I have no idea why the hash does not match up. Save your file so that it can be compared to the actual one.

For the next version, I recommend switching to codeload as that avoid this arcanery.

---

Do you want to switch for this version?  Otherwise, this is good to go.

---

I did a simple compare with the copy from codelaod and it has yet another hash.  So I unpacked both that one and the one I had in my dl/ dir, and the contents is identical AFAICS.  Then I repacked both of them on my laptop (which isn't my original build host), using
```
tar --numeric-owner --owner=0 --group=0 --sort=name -c conserver-8.2.4 | gzip -nc > a.tgz
tar --numeric-owner --owner=0 --group=0 --sort=name -c conserver-8.2.4 | gzip -nc > b.tgz
```
with the two conserver-8.2.4 dirs from respectively dl/ and codeload.  And whaddynow?  I ended up with identical tar files with the hash @cotequeiroz gave:

```
bjorn@miraculix:/tmp$ sha256sum v8.2.4 conserver-8.2.4.tar.gz {a,b}.tgz
8d8b263d9531ebb827257bd67baf9200cac4fb7ffbe033122686e39ca42cddd6  v8.2.4
b95658a2c8470da32759b65ed60e9d5747da3f84ca25f3be39f8aa0c53f4cc1d  conserver-8.2.4.tar.gz
8f02afb933d118fc4ccb73cda63213b76792dad4c080c7a64967a33371f8aa19  a.tgz
8f02afb933d118fc4ccb73cda63213b76792dad4c080c7a64967a33371f8aa19  b.tgz
```
This is fun...

The compressed files are of different size, so the hash differences aren't that surprising really:
```
bjorn@miraculix:/tmp$ ls -l v8.2.4 conserver-8.2.4.tar.gz {a,b}.tgz
-rw-r--r-- 1 bjorn bjorn 261817 Mar 27 20:34 a.tgz
-rw-r--r-- 1 bjorn bjorn 261817 Mar 27 20:34 b.tgz
-rw-r--r-- 1 bjorn bjorn 261828 Mar 27 20:32 conserver-8.2.4.tar.gz
-rw-r--r-- 1 bjorn bjorn 261682 Mar 27 20:29 v8.2.4
```
I think I'll just leave this for the smart people to figure out.  I don't really care that much.  Trusting a signed tag would be better.

---

From you description, and the fact that the `tar.gz` file is generated locally, it looks that your original build host's `gzip` is applying different compression than the rest of us.

---

> From you description, and the fact that the `tar.gz` file is generated locally, it looks that your original build host's `gzip` is applying different compression than the rest of us.

Yes, I thought so as well. But I decompressed them and found that the tar files where different too.

Anyway, thanks to both you and @neheb .  I decided to go for codeload to ensure some consistency.  Just needed a couple of attempts to get it right after fat fingering the first version

---

> Yes, I thought so as well. But I decompressed them and found that the tar files were different too.

Would you mind sharing the details? Or the original broken file? I want to figure out what went wrong on your host so that we can fix it in buildroot.

---

> For the next version, I recommend switching to codeload as that avoid this arcanery.

I recommend not using codeload as it does not guarantee long term availability of archives.

---

The differences are reproducible for me, so I created a new archive using
`PKG_SOURCE_PROTO:=git`
and put it on https://www.mork.no/~bjorn/.tmp-openwrt-dl-test/source-proto-git/ along with the Makefile I used and the output of 
`make -j1 V=s package/conserver/prepare`

For convenience I also put the codeload version in a separate directory, as well as repacked versions of both.  See the dirs under https://www.mork.no/~bjorn/.tmp-openwrt-dl-test/

The results above were replicated, even if I this time did the tar repacking on a different host.

My build host is running a current Debian stable with default tar and gzip packages:

```
bjorn@canardo:/usr/local/src/lede$ dpkg -l tar gzip
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                          Version             Architecture        Description
+++-=============================-===================-===================-===============================================================
ii  gzip                          1.6-5+b1            amd64               GNU compression utilities
ii  tar                           1.29b-1.1           amd64               GNU version of the tar archiving utility

bjorn@canardo:/usr/local/src/lede$ tar --version
tar (GNU tar) 1.29
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
bjorn@canardo:/usr/local/src/lede$ gzip --version
gzip 1.6
Copyright (C) 2007, 2010, 2011 Free Software Foundation, Inc.
Copyright (C) 1993 Jean-loup Gailly.
This is free software.  You may redistribute copies of it under the terms of
the GNU General Public License <http://www.gnu.org/licenses/gpl.html>.
There is NO WARRANTY, to the extent permitted by law.

Written by Jean-loup Gailly.
```